### PR TITLE
Adjust the release process: Sign off the cmd/ctl and push the tag only after merging the PR

### PR DESCRIPTION
**Pull request by wallrj**
**Preview**: https://deploy-preview-1363--cert-manager-website.netlify.app/docs/contributing/release-process/#process-for-releasing-a-version

While releasing v1.13.3, I didn't signoff the commit which updates the `cmd/ctl/go.mod` file, 
which caused the PR to fail the DCO checks.
 * I've added the `--signoff` option in this PR.

The problem was compounded by the fact that I had already pushed the `cmd/ctl@v1.13.3` tag to the cert-manager repo.
* I've  changed the order of the instructions so that the Tag only gets pushed after the PR has been merged.

### CODE REVIEW

**Comment by maelvls** [comment]
Hey. Thanks for updating the release process! And sorry about the pain created by my missing `--signoff` as well as the backwards instructions (push tag before merging PR...)

/lgtm
/approve

File: content/docs/contributing/release-process.md
Comment (wallrj) [inline content/docs/contributing/release-process.md:420]: Without the quotes, the files didn't get added.

```sh
$ # Must be run from the cert-manager repo folder.
cd cmd/ctl
go get github.com/cert-manager/cert-manager@$RELEASE_VERSION
cd ../..

find . -name go.mod -not -path ./_bin/\* -exec dirname '{}' \; | xargs -L1 -I@ sh -c 'cd @; go mod tidy'
git add **/go.mod **/go.sum
git commit --signoff -m"Update cmd/ctl's go.mod to $RELEASE_VERSION"
xargs: warning: options --max-lines and --replace/-I/-i are mutually exclusive, ignoring previous --max-lines value
On branch update-cmd/ctl/v1.12.7
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   cmd/ctl/go.mod
        modified:   cmd/ctl/go.sum
        modified:   test/integration/go.mod

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .envrc
        cert-manager/
        release-notes.md

no changes added to commit (use "git add" and/or "git commit -a")
```

**Comment by wallrj** [comment]
@maelvls PTAL

File: content/docs/contributing/release-process.md
Comment (maelvls) [reply content/docs/contributing/release-process.md:420]: My fault: I forgot that ** is specific to zsh. I didn’t realize that git add also supported ** 😅

**Review by maelvls — APPROVED** [review]
Thanks for the fixes and for the note about the Homebrew formula. 

/lgtm
/approve

**Pull request merged**
